### PR TITLE
feat: allow configuring keymap descriptions

### PR DESCRIPTION
**Pull request by snoblenet**
Allows the optional configuration of a keymap's `desc` alongside its `lhs`.

Related issues:

https://github.com/nvim-orgmode/orgmode/issues/788
https://github.com/nvim-orgmode/orgmode/issues/781

### CODE REVIEW

File: lua/orgmode/config/mappings/map_entry.lua [hunk 1]
```diff
@@ -106,6 +106,12 @@ function MapEntry:attach(default_mapping, user_mapping, opts)
     map_opts.prefix = nil
   end
 
+  if user_mapping ~= nil then
+    if user_mapping.desc ~= nil then
+      map_opts.desc = user_mapping.desc
+    end
+  end
```
Comment (kristijanhusak) [inline lua/orgmode/config/mappings/map_entry.lua:113]: We can remove some nesting here.


```suggestion
  if type(user_mapping) == 'table' and user_mapping.desc then
    map_opts.desc = user_mapping.desc
  end
```

**Review by kristijanhusak — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks great, just one small nit and it's good to go!

**Comment by snoblenet** [comment]
> Thanks for the PR! It looks great, just one small nit and it's good to go!

Cheers. I'm pretty sure I've now applied your requested change and merged in the latest master...?

**Review by kristijanhusak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**